README.md

```diff
@@ -1,229 +1,894 @@
+format. See [third example](#peering-deep-into-functions-with-lbr) for a demo
+of LBR mode.
+
+Here we'll just note that LBR mode allows customizing what kind of records CPU
+is instructed to record. It can be one of the following values: `any`,
+`any_call`, `any_return` (default), `cond, `call`, `ind_call`, `ind_jump`,
+`call_stack`, `abort_tx`, `in_tx`, `no_tx`. See Linux's
+[perf_event.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/perf_event.h#L180-L208)
+UAPI header and its `enum perf_branch_sample_type_shift`'s comments for brief
+description of each of those modes.
+
+By default, `retsnoop` assumes `any_return` LBR configuration, which records
+last N returns from functions. This is very useful for the second type of cases
+when we want to know which functions are being called without having to study
+code thoroughly. LBR stack will point to functions called before the deepest
+explicitly traced by `retsnoop` function was called, even if they weren't part
+of entry/non-entry set of functions. So you can keep iterating with `retsnoop`
+and expanding entry/non-entry function sets this way.
+
+For the first class of scenarios (complicated functions with multiple error
+conditions), LBR config `any` might be more appropriate. It records any
+function calls, returns, and jumps, both conditional and unconditional. It
+might help pinpoint exact statement *within a single big function* that is
+returning an error, as it's not bound to function boundaries.
+[LBR example](#peering-deep-into-functions-with-lbr) below is using this one to
+pinpoint the exact error condition.
+
+Note that LBR is a tricky business and it might not capture enough relevant
+details or sometimes it capture irrelevant details. If the latter is the case,
+you can trim it down with `--lbr-max-count` argument to emit specified number
+of most relevant entries.
+
+## Additional filters
+
+By default, `retsnoop` records any function call traces (based on entry and
+non-entry function sets) that result in triggered entry function returning
+error. The "error return" is defined heuristically and is either `NULL` for
+poitner-returning functions or `-Exxx` small negative error for integer
+returning one. This can be further adjusted, as you'll see below. But such
+function traces is collected globally across any process, which might be
+inconvenient and lead to irrelevant "spam" in output.
+
+`retsnoop` allows to adjust conditions under which it captures traces, though.
+
+### Error filters
+
+`-S` (`--success-stacks`) forces `retsnoop` to capture any function trace,
+disabling default logic of capturing only error-returning cases.
+
+`retsnoop` also allows to fine-tune which return results are considered to be
+erroneous with `-x` (`--allow-errors`) and `-X` (`--deny-errors`) arguments.
+They expect  either `Exxx`/`-Exxx` symbolic error codes (you can find all
+recorgnized by `retsnoop` errors
```

Comment:
s/recorgnized/recognized

In general I'd rephrase this as: (you can find all errors that are recognized by `retsnoop` [in this table](https://github.com/anakryiko/retsnoop/blob/master/src/utils.c#L10-L50)